File: lingting-spring-grpc/src/main/java/live/lingting/spring/grpc/mapstruct/SpringGrpcMapstruct.java
package live.lingting.spring.grpc.mapstruct;

import live.lingting.framework.grpc.properties.GrpcClientProperties;
import live.lingting.framework.grpc.properties.GrpcServerProperties;
import live.lingting.spring.grpc.properties.GrpcSpringProperties;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

/**
 * @author lingting 2024-02-05 16:18
 */
@Mapper
public interface SpringGrpcMapstruct {

	SpringGrpcMapstruct INSTANCE = Mappers.getMapper(SpringGrpcMapstruct.class);

	GrpcClientProperties client(GrpcSpringProperties properties, GrpcSpringProperties.Client client);

	GrpcServerProperties server(GrpcSpringProperties properties, GrpcSpringProperties.Server server);

}

File: lingting-spring-grpc/src/main/kotlin/live/lingting/spring/grpc/mapstruct/SpringGrpcMapstruct.kt
package live.lingting.spring.grpc.mapstruct

import live.lingting.spring.grpc.properties.GrpcSpringProperties
import org.mapstruct.Mapper

/**
 * @author lingting 2024-02-05 16:18
 */
@Mapper
interface SpringGrpcMapstruct {
    fun client(properties: GrpcSpringProperties, client: GrpcSpringProperties.Client): GrpcClientProperties

    fun server(properties: GrpcSpringProperties, server: GrpcSpringProperties.Server): GrpcServerProperties

    companion object {
        val INSTANCE: SpringGrpcMapstruct = Mappers.getMapper(SpringGrpcMapstruct::class.java)
    }
}
